File: product_managment_ui/src/service/productService.js
import axios from 'axios';  
// connection with backend
const API_URL = "http://localhost:5000";


class ProductService {
    saveProduct(product) {
        return axios.post(API_URL + "/saveProduct", product);
    }

    getAllProduct() {
        return axios.get(API_URL + "/");
    }

    getProductById(id) {
        return axios.get(API_URL + "/"+id);
    }

    deleteProduct(id) {
        return axios.get(API_URL + "/deleteProduct/"+id);
    }

    editProduct(product) {
        return axios.post(API_URL + "/editProduct/" + product.id, product);
    }
}

export default new ProductService;

File: product_managment_system/src/main/java/com/example/sohit/product_managment_system/service/ProductService.java
package com.example.sohit.product_managment_system.service;

import java.util.List;

import com.example.sohit.product_managment_system.model.Product;

public interface ProductService {

    public Product saveProduct(Product product);

    public List<Product> getAllProduct();

    public Product getProductById(Integer id);

    public String deleteProduct(Integer id);

    public Product editProduct(Product product, Integer id);



}
